app/authorization/server/functions/canAccessRoom.js

```diff
@@ -23,6 +24,13 @@ export const roomAccessValidators = [
 			return true;
 		}
 	},
+	async function(room, user) {
```

Comment:
why not  
```javascript
async function(room, { type }) { return type === 'app'; }

Reply:
sometimes, it passes `Meteor.user()` which only contains `_id` and `usernmae` property.

Reply:
here:
https://github.com/RocketChat/Rocket.Chat/blob/8cd4b0f1266d9bcfae58e9bfaa22284c7568055a/app/file-upload/server/methods/sendFileMessage.js#L16

Reply:
I realized we can avoid this unnecessary query since there's one in `Meteror.Methods({ canAccessRoom(...).. })`. I added a condition statement here
https://github.com/RocketChat/Rocket.Chat/blob/4b87683074afa84b71b99a6ad8a579a4c6b0b1f6/server/methods/canAccessRoom.js#L16-L24

to make sure the entire `user` object can be passed to the real `canAccessRoom` function on the server-side.

@ggazzo , is this a proper way for us to go?

Reply:
Mark it as resolved since we're going to remove the apps bypass validator inside canAccessRoom. See discussion https://open.rocket.chat/group/rocketchat-dev?msg=gPiDFesgW7x2p9Sj7 for details.